server/src/api/v1.py

```diff
@@ -100,13 +100,17 @@ def has_attachments(data: dict) -> bool:
     )
 
 
-def check_token_priority_permission(
+def check_token_permissions(
```

Comment:
The logic here is not too bad yet, but it's getting harder to follow. It's an important piece to get right, so we really want to make sure that it's testable and conceptually obvious.
Can we find a way to break this up so that it goes through a series of independent checks? For instance, one method check that either priority is 0 or it has permission to set the higher priority. The next method checks to see if it is either using an unrestricted queue, or it has permission to use the queue it asked for. There will be other checks in the future, such as whether it's using vault paths that it has access to, etc.
If it passes all of them, then the token is ok. Otherwise we reject it as soon as we see something is wrong.